pkg/readers/adapters/amqp.go

```diff
@@ -241,37 +251,35 @@ func NewAMQPAdapter() (ReadAdapter, error) {
 	}, nil
 }
 
-func NewAmqpConfig() (amqpAdapter.Config, error) {
-	// @TODO this needs to be extracted to its own method
+func generateConsumerID() (string, error) {
 	consumerId := os.Getenv("CONSUMER_ID")
 	if consumerId == "" {
 		// Work out consumer ID based on hostname: useful for k8s resources (pods controlled by deployment, statefulset)
 		hostname, err := os.Hostname()
 		if err != nil {
-			klog.Errorf("Could not set ConsumerID: %v", err)
+			return "", fmt.Errorf("could not get hostname for ConsumerID: %w", err)
 		}
 		pos := strings.LastIndex(hostname, "-")
 		consumerId = hostname[pos+1:]
+		if consumerId == "" {
+			return "", fmt.Errorf("could not extract consumer ID from hostname: %s", hostname)
+		}
 	}
 
-	// @TODO This needs to be extracted to its own object
-	data := struct {
-		ConsumerId string
-	}{
-		consumerId,
-	}
-
-	tpl := template.New("queueName")
-	tpl, err := tpl.Parse(helpers.GetEnv("RABBITMQ_QUEUE_NAME", ""))
-	if err != nil {
-		klog.Errorf("Invalid RABBITMQ_QUEUE_NAME: %v", err)
-	}
-	var buf bytes.Buffer
-	err = tpl.Execute(&buf, data)
+	return consumerId, nil
+}
+func NewAmqpConfig() (amqpAdapter.Config, error) {
+	consumerId, err := generateConsumerID()
 	if err != nil {
-		klog.Errorf("Invalid RABBITMQ_QUEUE_NAME: %v", err)
+		return amqpAdapter.Config{}, fmt.Errorf("could not generate ConsumerID: %w", err)
 	}
-	queueName := buf.String()
+	klog.Infof("Generated consumer ID: %s", consumerId)
+
+	queueName := strings.ReplaceAll(
```

Comment:
Not sure but maybe using template was a little bit complicated for this case. What about using strings ..